src/sparse/KokkosSparse_sptrsv_supernode.hpp

```diff
@@ -493,10 +503,16 @@ generate_supernodal_graph(bool col_major, graph_t &graph, int nsuper, const inpu
         nblocks ++;
         // count blocks per row for col_major
         hr (s2+1) ++;
+        // keep track of non-zero block ids
+        idxs (nidxs) = s2;
+        nidxs ++;
       }
     }
     // reset check
-    Kokkos::deep_copy (check, 0);
+    //Kokkos::deep_copy (check, 0);
+    for (size_type i = 0; i < nidxs; i++) {
+      check (idxs(i)) = 0;
```

Comment:
Consider making `check` an array of 64bit integers and using bit-wise operators to toggle bits. In this way, it may be reasonable to allocate `check` to be large enough so that it does not need to be overwritten before every iteration of the inner loop.

Reply:
What is the size of `check` in bytes now? What is the maximum value of `n`?

Reply:
It is the size of the local matrix and is an input from a user. So, it is difficult to say the max. Maybe, typically, it is around O(10^5)?

Reply:
If `n` is O(10^5), then `check` would need to be up to ~1GB if each element of `check` is a single bit? To reduce memory usage, a fall-back approach could be to have two or more limited-size `check` arrays; when one is used up, zero the used one out in parallel; this would require some synchronization.

Reply:
Thank you, @e10harvey. Any saving in the memory is good (though nsup is usually much smaller than n, and this setup is still done on the host).